client-java/src/unit-tests/java/uk/gov/gchq/palisade/client/internal/impl/ConfigurationTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.client.internal.impl;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import uk.gov.gchq.palisade.client.ClientException;
+
+import java.net.URI;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+class ConfigurationTest {
+
+    private static final String FILENAME = "palisade-client.yaml";
+
+    private static Configuration configuration;
+
+    @BeforeAll
+    static void setupAll() {
+        configuration = Configuration
+            .create(Map.of("service.url", "pal://localhost:8081/cluster?wsport=8082"));
```

Comment:
Suggested removing most of these url configurations

Reply:
This is the main URL provided by the user of the client. From this, the Palisade and Filtered Resource URIs are derived. Implementation and architectural details are abstracted from the user.